### PR TITLE
Credscan the entire repo

### PR DESCRIPTION
**Pull request by sima-zhu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Comment by sima-zhu** [comment]
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1513274&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=9036d097-2774-5136-505e-b4dc0a37f42d

**Comment by sima-zhu** [comment]
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1516538&view=results

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml [hunk 1]
```diff
@@ -65,6 +65,9 @@ jobs:
         VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
         Package.EnableSBOMSigning: true
       steps:
+        - template: /eng/common/pipelines/templates/steps/credscan.yml
```
Comment (danieljurek) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:68]: Is it OK if this runs in CI in public?

Comment (sima-zhu) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:68]: The credscan team page does not limit us for internal use. The log did not expose our internal information out. We currently have this setup in public ci for all other 4 langs. It is good to go.

**Comment by danieljurek** [comment]
Credscan results -- https://dev.azure.com/azure-sdk/public/_build/results?buildId=1594399&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=9036d097-2774-5136-505e-b4dc0a37f42d

**Comment by RickWinter** [comment]
@sima-zhu Can you complete this PR and enable the process?  If this is no longer applicable work, the PR should be closed.

**Comment by sima-zhu** [comment]
Pre-requisite:
https://github.com/Azure/azure-sdk-for-cpp/issues/3837

@RickWinter I will enable the PR validation once the errors get cleanup. :)

**Pull request closed without merging**